### DISCUSSION TITLE
Encoding Issue?

### DISCUSSION THREAD

**Discussion by mattbaylor** [postalserver/postal#1541] [Help with using Postal] [answered]
I'm attempting to get Postal to deliver messages generated in a couple of .NET apps.  I'm using standard SMTP relaying (not encrypting the connection currently). The delivered message appears to be encoded incorrectly:

![image](https://user-images.githubusercontent.com/2577926/129108739-bb62c438-1b5a-4a34-af28-1dfde7921cb8.png)

Through the postal interface, the HTML version of the message also appears encoded oddly:

![image](https://user-images.githubusercontent.com/2577926/129108819-09b6478b-0220-4c8e-b04e-9ae52cf54e78.png)

The source of the message looks more promising, which is leading me to believe it's an encoding problem:

![image](https://user-images.githubusercontent.com/2577926/129109324-0fe3896c-b27a-4fbd-88fa-2fa1d1bdaecd.png)

**Comment by willpower232** [comment]
I notice the header `Content-Transfer-Encoding: base64` but the email is actually plain text so that might be where things are getting confused.

**Reply by mattbaylor** [reply]
Since it's just an smtp transfer from the application, is there a place in postal to set a default encoding?

**Reply by willpower232** [reply]
You would have to look at how the message is generated by your application as Postal won't be doing too much to your messages unless you enable tracking.

**Reply by mattbaylor** [reply]
I am enabling tracking. I'll turn that off and see if the behavior changes.

**Reply by mattbaylor** [reply]
Turning off tracking did indeed clean up the encoding on the mail message. Where might I look to fix the encoding issue and still enable the tracking feature?

**Comment by mattbaylor** [comment, marked as answer]
Turning off the fast server for tracking fixed the encoding issues I was experiencing, however I really need tracking to work for my use case. Is this something that should be turned into an issue?

**Reply by willpower232** [reply]
I think this is a known issue #1442

**Reply by mattbaylor** [reply]
yup, thanks for redirecting me there.

**Comment by taylorbrooks** [comment]
I feel like the answer lies in these line(s) of code:

https://github.com/postalserver/postal/blob/main/lib/postal/message_parser.rb#L94-L130
https://github.com/postalserver/postal/blob/main/lib/postal/message_db/message.rb#L470-L475

**Comment by alglyzin** [comment]
It looks like the error is repeated when the domain is not used for tracking. Please fix it!

**Reply by uhlhosting** [reply]
You are also free to submit a PR.

**Reply by alglyzin** [reply]
What do you mean?

**Reply by uhlhosting** [reply]
This is a open source software, so people can fork it, improve it and submit a PR to be merged into the main repository.
